### PR TITLE
Store task errors following migrations and display them

### LINKED ISSUE #13242: GitHub migration: Show "API rate limit of 5000 still exceeded" to user
- Gitea version (or commit ref): 1.13.0+rc1
- Git version: 2.20.1
- Operating system: debian stable 10.6
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [X] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [X] i get migration failed too, but can't look into your logs ;)

## Description

When i try to migrate a big GitHub repo, i see that the API limit is reached in the logs, but as an end-user, i see just "migration failed" without reason.

Please show it also to the user and what they can do.

> API rate limit of 5000 still exceeded until 2020-10-21 16:33:28 +0200 CEST

Just wait and try again.

Related to https://github.com/go-gitea/gitea/issues/13241

## Log

`modules/task/task.go:51:handle() [E] Run task failed: GET https://api.github.com/repos/NixOS/nixpkgs: 403 API rate limit of 5000 still exceeded until 2020-10-21 16:33:28 +0200 CEST, not making remote request. [rate reset in 34m49s]`

## Screenshots

Screenshot from 2020-10-21 17-31-45

### PR DESCRIPTION
**Pull request by zeripath**
When migrate tasks fail store the error in the task table
and ensure that they show on the status page.

Fix #13242

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Review by mrsdizzie — APPROVED** [review]

File: templates/repo/migrate/migrating.tmpl [hunk 1]
```diff
@@ -7,11 +7,16 @@
...
 					<div class="ui stackable middle very relaxed page grid">
-						<div id="repo_migrating" class="sixteen wide center aligned centered column" repo="{{.Repo.Repository.FullName}}">
+						<div id="repo_migrating" class="sixteen wide center aligned centered column" task="{{.MigrateTask.ID}}">
 							<div>
 								<img src="{{StaticUrlPrefix}}/img/loading.png"/>
 							</div>
 						</div>
+						<div id="repo_migrating_failed_image" class="sixteen wide center aligned centered column">
+							<div>
+								<img src="{{StaticUrlPrefix}}/img/failed.png"/>
```
Comment (lunny) [inline templates/repo/migrate/migrating.tmpl:17]: Is this a new image?

File: web_src/js/index.js [hunk 2]
```diff
@@ -192,33 +192,42 @@ function updateIssuesMeta(url, action, issueIds, elementId) {
...
-            if (xhr.responseJSON.status === 0) {
+            if (xhr.responseJSON.status === 4) {
               window.location.reload();
               return;
+            } else if (xhr.responseJSON.status === 3) {
+              $('#repo_migrating_progress').hide();
+              $('#repo_migrating').hide();
+              $('#repo_migrating_failed').show();
+              $('#repo_migrating_failed_image').show();
+              $('#repo_migrating_failed_error').html(xhr.responseJSON.err);
```
Comment (silverwind) [inline web_src/js/index.js]: Is that error HTML? If not, use `.text()`

File: templates/repo/migrate/migrating.tmpl (hunk 1 above)
Comment (zeripath) [reply templates/repo/migrate/migrating.tmpl:17]: yes. It's simply the gitea cup upside down

File: web_src/js/index.js (hunk 2 above)
Comment (zeripath) [reply web_src/js/index.js]: Oops!

```suggestion
              $('#repo_migrating_failed_error').text(xhr.responseJSON.err);
```

**Comment by 6543** [comment]
on migration screen loading you can see the broken for a split of a second before normal progress icon is shown

**Comment by zeripath** [comment]
OK try now

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
Backported in #13287

**Pull request merged**